src/inheritAttributes.js

```diff
@@ -431,8 +500,35 @@ export const inheritAttributes = (mpd, options = {}) => {
     mpdAttributes.locations = locations.map(getContent);
   }
 
+  // Convert to PeriodInformation objects since missing attributes will be added, and it
+  // is better to add them to attributes than to add them to the Node to be parsed later.
+  const periods = periodNodes.map((node) => {
+    return {
+      node,
+      attributes: parseAttributes(node)
+    };
+  });
+  // Since toAdaptationSets acts on individual period objects right now, the simplest
+  // approach to adding properties that require looking at prior periods is to add them
+  // before toAdaptationSets is called. If more such properties are added, it may be
+  // better to refactor toAdaptationSets.
+  const periodsPlusMissingAttributes = [];
+
+  periods.forEach((period, index) => {
```

Comment:
Should we merge these two loops? Doesn't seem like we need to loop over the same thing separate times.